colossalai/nn/layer/moe/experts.py

```diff
@@ -144,29 +84,57 @@ def forward(self, inputs):    # inputs [g, el, c, h]
         outputs = outputs.transpose(0, 1).contiguous()
         return outputs
 
+    def state_dict(self, destination=None, prefix='', keep_vars=False):
+        dp_rank = dist.get_rank(get_dp_group(self))
+        ep_rank = dist.get_rank(get_ep_group(self))
+        ep_size = get_ep_size(self)
+        # dp rank 0 will save the state dict
+        if dp_rank == 0:
+            for name, module in self.named_parameters():
```

Comment:
Is it a typo?

Reply:
if you refer to ep, that means expert parallel. There is only tp process group, I use it to represent ep

Reply:
done